metrics-scraper/monitoring.go

```diff
@@ -69,80 +70,146 @@ func Start(configPath string) {
-		return err
+		errs = append(errs, err)
 	}
 
-	return nil
+	if len(errs) > 0 {
+		mezodUpGauge.WithLabelValues(moniker, networkID).Set(0)
+	} else {
+		mezodUpGauge.WithLabelValues(moniker, networkID).Set(1)
+	}
+
+	return errors.Join(errs...)
 }
 
-func latestBlockAndTimestamp(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+func latestBlockAndTimestamp(ctx context.Context, client *rpc.Client, moniker, networkID string) (err error) {
+	defer func() {
+		if err != nil {
+			// set latest block to 0 for error
+			mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(0)
+			// set latest timestamp to 0 for error
+			mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(0)
+		}
+	}()
+
 	result := struct {
 		Number    string `json:"number"`
 		Timestamp string `json:"timestamp"`
 	}{}
-	err := client.CallContext(ctx, &result, ethGetBlockByNumberEndpoint, "latest", false)
+
+	err = client.CallContext(ctx, &result, ethGetBlockByNumberEndpoint, "latest", false)
 	if err != nil {
-		mezodUpGauge.WithLabelValues(moniker, networkID).Set(0)
 		return fmt.Errorf("couldn't call %v for %v: %v", ethGetBlockByNumberEndpoint, moniker, err)
 	}
 
-	mezodUpGauge.WithLabelValues(moniker, networkID).Set(1)
-
+	errs := []error{}
 	latestBlock, err := strconv.ParseUint(strings.TrimPrefix(result.Number, "0x"), 16, 64)
 	if err != nil {
-		return err
+		errs = append(errs, fmt.Errorf("invalid latestBlock: %v", err))
+	} else {
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(float64(latestBlock))
 	}
-	mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(float64(latestBlock))
 
 	ts, err := strconv.ParseUint(strings.TrimPrefix(result.Timestamp, "0x"), 16, 64)
 	if err != nil {
-		return err
+		errs = append(errs, fmt.Errorf("invalid timestamp: %v", err))
+	} else {
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(float64(ts))
 	}
-	mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(float64(ts))
 
-	return nil
+	return errors.Join(errs...)
 }
 
-func nodeVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+func nodeVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) (err error) {
+	defer func() {
+		if err != nil {
+			// set it to a valid semver showing nicely there's an error
+			mezodVersionGauge.WithLabelValues(moniker, networkID, "0.0.0-unknown").Set(1)
+		}
+	}()
+
+	// always remove the previous metrics to ensure that there's no duplicates when
+	// the labels version changes
+	mezodVersionGauge.DeletePartialMatch(map[string]string{
+		"moniker":  moniker,
+		"chain_id": networkID,
+	})
+
 	var result string
-	err := client.CallContext(ctx, &result, web3ClientVersionEndpoint)
+	err = client.CallContext(ctx, &result, web3ClientVersionEndpoint)
 	if err != nil {
 		return fmt.Errorf("couldn't call %v for %v: %v", web3ClientVersionEndpoint, moniker, err)
```

Comment:
Let's be consistent:
```suggestion
        err = fmt.Errorf("couldn't call %v for %v: %v", web3ClientVersionEndpoint, moniker, err)
		return 
```
Same for the error returned for `if len(segments) != 4` below.

Reply:
addressed in: 94e0597d570fbb1a694bc047ed7b89d8570a1289